src/mercury/Mercury.ts

```diff
@@ -129,10 +129,14 @@ export default class Mercury implements MercuryInterface {
    */
   async sendMessageParseMessage(
     message: Domain.Message
-  ): Promise<Domain.Message> {
-    const responseBody = await this.sendMessage<any>(message);
-    const responseJSON = JSON.stringify(responseBody);
-    return this.unpackMessage(responseJSON);
+  ): Promise<Domain.Message | undefined> {
+    try {
+      const responseBody = await this.sendMessage<any>(message);
+      const responseJSON = JSON.stringify(responseBody);
+      return await this.unpackMessage(responseJSON);
+    } catch (err) {
+      return undefined
+    }
```

Comment:
please can you explain this? not sure we should be swallowing an error here?

Reply:
I've now slightly changed the function, the goal is to avoid throwing an error if the response on delivery is null. But this may affect also when the endpoints are down or also return other exceptions. 

The code is changed now so that it only returns null if decoding the response message fails with an exception, if the request fails it will now not hide that and fail.